src/main/java/ca/wolfram/beta/symath/operations/AddNode.java

```diff
@@ -62,7 +92,18 @@ public double operationEval(VMap map) {
     public String toString() {
         StringBuilder s = new StringBuilder();
         s.append("(");
-        getChildren().forEach((c) -> s.append(c).append(" + "));
-        return s.substring(0, s.length() - 3) + ")";
+        boolean isFirst = true;
+        for (MathNode c : getChildren()) {
+            if (isFirst) {
+                s.append(c);
+                isFirst = false;
+                continue;
+            }
+            if (c.getType() == NodeType.NEGATE || (c.getType() == NodeType.CONSTANT) && c.eval(null) < 0)
```

Comment:
even though we know the order of operation for the boolean operators. i think it is better practice to have brackets to prevent confusion